src/lib.rs

```diff
@@ -153,6 +153,11 @@ impl TlsConnector {
             session,
         }))
     }
+
+    /// Get a read-only reference to underlying config
+    pub fn config(&self) -> Arc<ClientConfig> {
```

Comment:
We should avoid cloning here, as the caller can do that if they need it -- just yield `&Arc<ClientConfig>` instead. (Same for both.)

Reply:
Thank you. Just updated this part.